packages/@storefront/core/test/unit/services/tracker.ts

```diff
@@ -190,12 +190,49 @@ suite('Tracker Service', ({ expect, spy, stub, itShouldExtendBaseService }) => {
 
       expect(override).to.be.calledWithExactly(value, currentEvent);
     });
+
+    it('should call attachGbiEventMetadata() with override', () => {
+      const event: any = { a: 'b' };
+      const currentEvent = { c: 'd' };
+      const result = { e: 'f', metadata: [{ i: 'j' }] };
+      const override = spy(() => result);
+      const value = { g: 'h' };
+      const attachGbiEventMetadataSpy = spy();
+      service.attachGbiEventMetadata = attachGbiEventMetadataSpy;
+      stub(service, 'addMetadata').withArgs(event).returns(currentEvent);
+
+      service.buildEvent(override, event, value);
+
+      expect(attachGbiEventMetadataSpy).to.be.calledWith(result);
+    });
+  });
+
+  describe('attachGbiEventMetadata()', () => {
+    const GBI_EVENT = { key: 'gbi', value: 'true' };
+
+    it('should add the GBI_EVENT object to the override event metadata', () => {
+      const override = { e: 'f', metadata: [{ i: 'j' }] };
+
+      const overriddenEvent = service.attachGbiEventMetadata(<any>override);
```

Comment:
Is it doable to annotate `override` with the `any` type instead of casting it here?

Reply:
Will do. Marking as resolved.